feat: New Onboarding UI - Carousel and Terms of Use UI

ieow opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Update New SRP onboarding flow as per figma design
https://www.figma.com/design/pViOUcmjwhEzFsdrwknpNc/Onboarding-redesign?node-id=434-55200&t=wmEoQSF9g9lFtkwp-0

This PR is part 1 of 7 for the new SRP ui update

This PR covers
- new onboarding carousel ui
- terms and conditions in bottom sheet


<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Manual testing steps**

1. Open the App
2. Click on `Get Started` button
3. Agree to Terms of Use
4. Close app and re-open
5. Terms of Use popup shouldn't appear again


## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

https://github.com/user-attachments/assets/b4b63f58-49ad-4561-a7e6-dfe8922bad0b



<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

ieow committed:
feat: new onboarding carousel and use terms

Co-authored-by: smgv <gpatra1996@gmail.com>

smgv committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

chaitanyapotti commented on app/component-library/components/Modals/ModalMandatory/ModalMandatory.styles.ts:
```diff
@@ -50,26 +69,25 @@ const styleSheet = (params: { theme: Theme }) => {
       color: colors.text.alternative,
       ...(typography.sBodySM as TextStyle),
       fontFamily: getFontFamily(TextVariant.BodySM),
+      marginBottom: Platform.OS === 'ios' ? 8 : 16,
```
Use platform.select and use the same value for ios and macos. All normal ios apps can now run on macos

chaitanyapotti commented on app/components/Views/OnboardingCarousel/index.tsx:
```diff
@@ -214,12 +242,20 @@ export const OnboardingCarousel: React.FC<OnboardingCarouselProps> = ({
     updateNavBar();
   }, [colors, updateNavBar]);
 
+  const getBackgroundColor = () => {
```
useMemo here

ieow committed:
fix: comment

smgv committed:
Fix: ModalMandatory Test Case

ieow committed:
fix: address comment
update to use 8 for macos, and rename variables

chaitanyapotti reviewed (dismissed):
lgtm

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Onboarding Carousel Test case Update

smgv committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' of https://github.com/MetaMask/metamask-mobile into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Onboarding Carousel Change Tab Test Case Added

smgv committed:
Fix: Onboarding Carousel Update ChangeTab

smgv committed:
Fix: Navbar getOnboardingCarouselNavbarOptions Test Case

smgv committed:
Fix: Test Case for OnBoardingScreenWithBg and TermsOfUse added

smgv committed:
Fix: ModalMandatory Test Case Updated

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Test case update for PR 1.1

smgv committed:
Merge branch 'feat/srp-onboarding-carousel-tnc' of https://github.com/MetaMask/metamask-mobile into feat/srp-onboarding-carousel-tnc

smgv committed:
Fix: Code Fix nested ternary operation

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
test: fix e2e acceptTermOfUse right after tap Started

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
test: move acceptTermOfUse to right point

hieu-w committed:
test: skip wrong e2e implement

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

hieu-w committed:
Revert "test: skip wrong e2e implement"

This reverts commit 88015e32bc491805ee6cdc4bb39b5c9ebe304e94.

hieu-w committed:
test: e2e quarantine send-eth test

tommasini commented on app/component-library/components/Modals/ModalMandatory/ModalMandatory.test.tsx:
```diff
@@ -1,50 +1,236 @@
+    // Press button
+    fireEvent.press(button);
+
+    // expect(mockOnAccept).toHaveBeenCalled();
```
Should we uncomment this line?

ieow replied:
uncommented and fixed the issue.

brianacnguyen commented on app/component-library/components/Modals/ModalMandatory/ModalMandatory.styles.ts:
```diff
@@ -50,26 +69,29 @@ const styleSheet = (params: { theme: Theme }) => {
+      width: 40,
+      height: 40,
+      borderRadius: 40 / 2,
+      backgroundColor: commonColors.modalScrollButton,
```
this will mean that the color will not be themed. is this intentional?

brianacnguyen replied:
@smgv Can you link me to the design with this color? I'll consult with our DS designer

tommasini replied:
I think it's here https://www.figma.com/design/pViOUcmjwhEzFsdrwknpNc/Onboarding-redesign?node-id=183-30163&p=f&t=Q6mM5lnOkmr7Qo0G-0

brianacnguyen replied:
you can use `colors.primary.muted` for this instead and you'll get a themed color youre looking for

smgv committed:
Fix: Comment Review

hieu-w committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

ieow committed:
feat: New welcome screen (#15666)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Update New SRP onboarding flow as per figma design

https://www.figma.com/design/pViOUcmjwhEzFsdrwknpNc/Onboarding-redesign?node-id=434-55200&t=wmEoQSF9g9lFtkwp-0

This PR is part 2 of 7 for the new SRP ui update

This PR covers

- new ui for welcome page

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. After opening app, agree to terms
2. Come to the Let's get started page

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

https://github.com/user-attachments/assets/2e37d1a2-3587-42d5-b90a-9965b5168ed2


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Chaitanya Potti <chaitanya.potti@gmail.com>
Co-authored-by: smgv <gpatra1996@gmail.com>
Co-authored-by: hieu-w <maestrokoder@gmail.com>
Co-authored-by: hieu-w <hieu@tor.us>

tommasini reviewed (dismissed):
I've miss clicked and merge this PR. I'm good to merge it like this since I've merged this PR in https://github.com/MetaMask/metamask-mobile/pull/15666

EtherWizard33 commented:
Hi, in the video above I did not see the carousel but saw it when running it.
On android looked like this, it's as expected? 
nit: btw I see a black bar at the bottom and white up top, mentioning it in case its not expected.

| Carousel       |        |
|--------------|--------------|
| <img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/eef3edd1-4377-4933-9ad7-6726194e8a3f"> |<img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src=""> |

EtherWizard33 reviewed (commented):
ran the PR for review, and added a comment just above https://github.com/MetaMask/metamask-mobile/pull/15665#issuecomment-2922966583

chaitanyapotti committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc

ieow committed:
fix: transparent status bar

ieow committed:
fix: lint

ieow committed:
fix: apply for android only

ieow commented:
Fixed the bottom navigation bar ( full-screen indicator) here 
https://github.com/MetaMask/metamask-mobile/pull/15962
created as separate PR as it requires new expo packages

EtherWizard33 reviewed (dismissed):
pulled the branch ran the code, as in manual steps, worked as expected
left a comment about bottom and top bar background color in case it can be improved easily. LGTM approved

smgv committed:
Fix: Comment Review For Terms of use ScrollButton to use Floating Button design

EtherWizard33 reviewed (approved):
re-approved

tommasini commented:
As a note, this PR also includes this PR: https://github.com/MetaMask/metamask-mobile/pull/15666

tommasini committed:
Merge branch 'main' into feat/srp-onboarding-carousel-tnc
